ImputationPipeline/CreateAggregationSets/tests/CreateAggregationSetsTest.py

```diff
@@ -0,0 +1,192 @@
+import ImputationPipeline.CreateAggregationSets.CreateAggregationSets as CreateAggregationSets
+import unittest
+import os
+import firecloud.api as fapi
+import json
+import random
+
+current_dir = os.path.dirname(__file__)
+resources_dir = os.path.join(current_dir, "resources")
+
+
+def json_lists_to_sets(o):
+    if isinstance(o, list):
+        return {json_lists_to_sets(v) for v in o}
+    elif isinstance(o, dict):
+        return {k: json_lists_to_sets(v) for k, v in o.items()}
+    return o
+
+
+def build_sample_sets_lite(sample_sets: list):
+    sample_sets_lite = dict()
+    for sample_set in sample_sets:
+        name = sample_set["name"]
+        group = sample_set["attributes"]["group"]
+        lab_batch = sample_set["attributes"]["lab_batch"]
+        samples = {sample["entityName"] for sample in sample_set["attributes"]["samples"]["items"]}
+        delivered = sample_set["attributes"]["delivered"]
+        redeliver = sample_set["attributes"]["redeliver"]
+        sample_sets_lite[name] = {"group": group, "lab_batch": lab_batch, "samples": samples, "delivered": delivered,
+                                  "redeliver": redeliver}
+    return sample_sets_lite
+
+
+class IntegrationTests(unittest.TestCase):
+    workspace_namespace = None
+    workspace_name = None
+    maxDiff = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.workspace_namespace = os.getenv("WORKSPACE_NAMESPACE")
```

Comment:
Great idea about creating a test workspace!

Are you sure you want to get this from `WORKSPACE_NAMESPACE` though? If I were to execute this in a notebook in the ck_copy workspace for example (which lives in emerge_prod) it would create a workspace in emerge_prod. I don't know if you can give this setup method arguments, but perhaps you can change the environment variable name to `TEST_WORKSPACE_NAMESPACE` or something like that, so that users _have_ to set it manually

Reply:
yeah, good idea